pandas/tests/frame/test_convert_to.py

```diff
@@ -612,3 +612,19 @@ def test_to_dict_wide(self):
         result = df.to_dict("records")[0]
         expected = {"A_{:d}".format(i): i for i in range(256)}
         assert result == expected
+
+    def test_to_dict_orient_dtype(self):
+        # https://github.com/pandas-dev/pandas/issues/22620
+        # Input Data
+        input_data = {"a": [1, 2, 3], "b": [1.0, 2.0, 3.0], "c": ["X", "Y", "Z"]}
+        df = DataFrame(input_data)
+        # Expected Dtypes
+        expected = {"a": int, "b": float, "c": str}
+        # Extracting dtypes out of to_dict operation
+        df_dict = df.to_dict("records")[0]
```

Comment:
can you compare the entire result, not just the first record

Reply:
this is now done